src/runtime/auto-prompt-manager.ts

```diff
@@ -467,6 +501,110 @@ export class AutoPromptManager {
     return potentialAction;
   }
 
+  private async getTargetedInterventionOrchestration(
+    clientConfig: ClientConfig,
+    article: Article,
+    contentType: ContentType
+  ): Promise<InterventionOrchestration | void> {
+    let eligibleActions = article.audienceActions?.actions;
+    let targetedInterventions =
+      article.actionOrchestration?.interventionFunnel?.prompts;
+    if (
+      !eligibleActions ||
+      eligibleActions.length === 0 ||
+      !targetedInterventions ||
+      targetedInterventions.length === 0
+    ) {
+      return;
+    }
+
+    // Complete client-side eligibility checks for actions.
+    const actionsTimestamps = await this.getTimestamps();
+    eligibleActions = eligibleActions.filter((action) =>
+      this.checkActionEligibility_(action.type, actionsTimestamps!)
+    );
+    if (eligibleActions.length === 0) {
+      return;
+    }
+
+    // Filter the funnel of interventions by eligibility.
+    targetedInterventions = targetedInterventions.filter((intervention) =>
+      eligibleActions
+        .map((action) => action.configurationId)
+        .includes(intervention.configId)
+    );
+    if (targetedInterventions.length === 0) {
+      return;
+    }
+
+    if (contentType === ContentType.CLOSED) {
+      return targetedInterventions[0];
+    }
+
+    // Only other supported ContentType is OPEN.
+    let nextIntervention: InterventionOrchestration | undefined = undefined;
+    // Check Default FrequencyCapConfig is valid.
+    if (
+      !this.isValidFrequencyCap_(
+        clientConfig.autoPromptConfig?.frequencyCapConfig
+      )
+    ) {
+      this.eventManager_.logSwgEvent(
+        AnalyticsEvent.EVENT_FREQUENCY_CAP_CONFIG_NOT_FOUND_ERROR
+      );
+      return targetedInterventions[0];
+    }
+
+    for (const intervention of targetedInterventions) {
+      const frequencyCapDuration = this.getPromptFrequencyCapDuration_(
+        clientConfig.autoPromptConfig?.frequencyCapConfig!,
+        intervention
+      );
+      if (this.isValidFrequencyCapDuration_(frequencyCapDuration)) {
```

Comment:
Note: much of the computation for whether a prompt is FrequencyCapped (or if the flow is globally capped) is identical to the old flow.